File: src/main/java/com/jakeesveld/zoos/model/Telephone.java
package com.jakeesveld.zoos.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;

@Entity
@Table(name = "telephone")
public class Telephone {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long phoneid;

    private String phonetype;
    private String phonenumber;

    @ManyToOne
    @JoinColumn(name = "zooid")
    @JsonIgnoreProperties("zooid")
    private Zoo zoo;

    public Telephone() {
    }

    public Telephone(String phonetype, String phonenumber) {
        this.phonetype = phonetype;
        this.phonenumber = phonenumber;
    }

    public Long getPhoneid() {
        return phoneid;
    }

    public void setPhoneid(Long phoneid) {
        this.phoneid = phoneid;
    }

    public String getPhonetype() {
        return phonetype;
    }

    public void setPhonetype(String phonetype) {
        this.phonetype = phonetype;
    }

    public String getPhonenumber() {
        return phonenumber;
    }

    public void setPhonenumber(String phonenumber) {
        this.phonenumber = phonenumber;
    }

    public Zoo getZoo() {
        return zoo;
    }

    public void setZoo(Zoo zoo) {
        this.zoo = zoo;
    }
}

File: src/main/java/com/jakeesveld/zoos/model/Telephone.kt
package com.jakeesveld.zoos.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

import javax.persistence.*

@Entity
@Table(name = "telephone")
class Telephone {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var phoneid: Long? = null

    var phonetype: String? = null
    var phonenumber: String? = null

    @ManyToOne
    @JoinColumn(name = "zooid")
    @JsonIgnoreProperties("zooid")
    var zoo: Zoo? = null

    constructor() {}

    constructor(phonetype: String, phonenumber: String) {
        this.phonetype = phonetype
        this.phonenumber = phonenumber
    }
}

